ams/server/src/main/java/com/netease/arctic/server/persistence/SqlSessionFactoryProvider.java

```diff
@@ -109,38 +111,73 @@ public void init(Configurations config) {
 
   // create tables for derby database type
   private void createTablesIfNeed(Configurations config) {
-    if (ArcticManagementConf.DB_TYPE_DERBY.equals(config.getString(ArcticManagementConf.DB_TYPE))) {
+    String dbTypeConfig = config.getString(ArcticManagementConf.DB_TYPE);
```

Comment:
The codes here seem to have significant duplication, with only the SQL for checking if tables have already been initialized is different for different types of databases. The rest of the logic appears to be the same.

Can you see if you can refactor the code to reduce code duplication?